### ISSUE TITLE
TalkBack doesn't navigate to headings contained in StackLayout with a SemanticProperties.Description

### ISSUE THREAD

**Issue by gbarkerz** [dotnet/maui#8524]
### Description

I needed to create a page containing multiple headings, so set SemanticProperties.HeadingLevel on particular labels. I also needed to create nested groupings with nesting that's accessible to screen readers, and in order to achieve this, I set SemanticProperties.Description on some StackLayouts. This worked well on Windows with the Narrator screen reader, and I mentioned how useful this was at https://twitter.com/gbarkerZ/status/1543533603921006592?s=20&t=cRkhQlhf4LSAISDmZZjFEQ.

I then built it and ran on Android, and found that the TalkBack screen reader apparently won't navigate to any headings that lie in a StackLayout with a SemanticProperties.Description. So the necessary navigation by headings doesn't work.

My version of .NET is 6.0.400-preview.22301.10, 

### Steps to Reproduce

1. Build and run an Android version of the app at https://github.com/gbarkerz/MyVeryFirstMauiApp. The code of interest is in https://github.com/gbarkerz/MyVeryFirstMauiApp/blob/master/MyVeryFirstMauiApp/MainPage.xaml.

2. Turn on the TalkBack screen reader on the Android device, set TalkBack's reading control to navigate by headings, and move forward through the headings in the app.

3. TalkBack moves through most headings in the app, but bypasses the heading that's contained in the StackLayout with the SemanticProperties.Description. 

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 9

**Comment by jsuarezruiz** [comment]
Verified with [MyVeryFirstMauiApp-master.zip](https://github.com/dotnet/maui/files/9044305/MyVeryFirstMauiApp-master.zip)

**Comment by gbarkerz** [comment]
Based on further experiments, it seems that VoiceOver on Mac behaves the same way at the app as TalkBack does on Android, in that it bypasses headings contained in StackLayouts with a SemanticProperties.Description.

**Comment by Karan4495** [comment]
GitHubTags:
#.NET MAUI; #A11ySev1; #A11yMAS; #A11yTCS; #Benchmark; #Android; #WCAG2.1.1; #GH_.NETMAUI_Android_Apr2022; #AILimited​;

**Comment by PureWeen** [comment]
@gbarkerz @Karan4495 This isn't really an accessibility issue with .NET MAUI

If you remove the `ScrollView` from the MAUI app then it all works as @gbarkerz  describes. It looks like this is just how Android groups the behavior on Android

If you use the following code on Android you'll see the exact same behavior that .NET MAUI has

If you remove the `ScrollView` then talkback is unable to navigate to any of 

```XML
    <ScrollView
        android:layout_width="match_parent"
        android:layout_height="match_parent">
        <androidx.appcompat.widget.LinearLayoutCompat
            android:layout_width="wrap_content"
            android:contentDescription="I am the linear layout"
            android:importantForAccessibility="yes"
            android:layout_height="wrap_content">
            <TextView
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:accessibilityHeading="true"
                android:text="Hello there people"
                ></TextView>
            <TextView
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:text="wakka wakka"
                ></TextView>
            <TextView
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:accessibilityHeading="true"
                android:text="Hello there people"
                ></TextView>
        </androidx.appcompat.widget.LinearLayoutCompat>
    </ScrollView>
   ```

**Comment by merriemcgaw** [comment]
@PureWeen then does this indicate that it's not really a bug and can be resolved as such?

**Comment by PureWeen** [comment]
@merriemcgaw I'd say there are two options here

1) close as not a bug
2) lower the severity level and research to determine if it's a platform bug. I'm not sure at the Android level if this would be perceived as "By Design" or "Talk back bug" 

I lean towards option two. Keeping something like this around also gives anyone a place to indicate if they are running into this same issue. 

We can do some research here and log the bug with Android to see how they mark it.

**Comment by gbarkerz** [comment]
@PureWeen Thanks for the investigation, it's interesting to learn that this seems to be underlying Android behavior. It would be good to know how a related issue against Android would be considered. My original UI design seemed reasonable to me, ie having headings inside semantic groups, but at least if the current behavior doesn't change, I'll know to explicitly avoid designing this type of UI layout in an Android app in the future.

**Comment by PureWeen** [comment]
### Android App demonstrating the same behavior

![2022-10-20_10-10-45](https://user-images.githubusercontent.com/5375137/196987954-560565ac-0fa4-4212-b73c-9f9cc5337ecd.gif)

### Android App if you remove the scrollview
![2022-10-20_10-13-05](https://user-images.githubusercontent.com/5375137/196988396-5f2758e4-b19f-4023-b958-c01e50aa09f2.gif)

**Comment by rachelkang** [comment]
Workaround:  Add `AutomationProperties.IsInAccessibleTree="False"` to the ScrollView

**Comment by keerthiduvvuri** [comment]
Hi Shane 
The issue is Reproducing in the "My Application" 
Can you please provide us the sample app to log external bug to talk back.
Thanks.

**Comment by rperez030** [comment]
Hi @keerthiduvvuri . reposting the code example that @PureWeen  added in a previous comment.
```

    <ScrollView
        android:layout_width="match_parent"
        android:layout_height="match_parent">
        <androidx.appcompat.widget.LinearLayoutCompat
            android:layout_width="wrap_content"
            android:contentDescription="I am the linear layout"
            android:importantForAccessibility="yes"
            android:layout_height="wrap_content">
            <TextView
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:accessibilityHeading="true"
                android:text="Hello there people"
                ></TextView>
            <TextView
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:text="wakka wakka"
                ></TextView>
            <TextView
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:accessibilityHeading="true"
                android:text="Hello there people"
                ></TextView>
        </androidx.appcompat.widget.LinearLayoutCompat>
    </ScrollView>
```

**Comment by rachelkang** [comment]
> Workaround: Add `AutomationProperties.IsInAccessibleTree="False"` to the ScrollView

![Animation](https://user-images.githubusercontent.com/21988533/200002508-81d6c0d5-bf8d-4043-93ea-319a4030f1ab.gif)

**Comment by keerthiduvvuri** [comment]
Hi Shane
As per discussion with SME we have to reduce the severity of this bug to sev3, and I have raised a third party bug to talkback which is tracked by https://dev.azure.com/CSS-IRT/CSS%20IRT/_workitems/edit/202560
and we can close this bug(I am not having access to close this bug).
Thanks.
